Formatter stuff

vivianbp opened the pull request:
major work on the formatter

vivianbp committed:
wtf is a branch

vivianbp committed:
finished, time for testing

vivianbp committed:
updated readme and a few other things.

porglezomp commented on package.json:
```diff
@@ -36,21 +37,53 @@
+        "lc2k.format.noopoffset": {
+          "type": "integer",
+          "default": 3,
+          "description": "Offset for comments after a jalr, inline is 2."
```
Wrong description.

porglezomp commented on package.json:
```diff
@@ -36,21 +37,53 @@
+          "default": true,
+          "description": "Add a newline at the end of a file if there isn't one."
+        },
+        "lc2k.format.filloffset": {
```
Should these be styled like `fillOffset` instead?

porglezomp replied:
Mention the `-1` to disable that's described above?

porglezomp replied:
Maybe use `null` instead of `-1`, since lots of other config things seem to use `null` for that purpose.

porglezomp commented on package.json:
```diff
@@ -36,21 +37,53 @@
+          "default": true,
+          "description": "Enables this extention to format .as files."
+        },
+        "lc2k.format.addNewlineAtEOF": {
```
This could probably be replaced by adding to the language-specific settings both:
```json
  "files.trimFinalNewlines": true,
  "files.insertFinalNewline": true,
```

vivianbp replied:
did this! now I just have to change the readme

porglezomp commented on package.json:
```diff
@@ -36,21 +37,53 @@
+        "lc2k.format.filloffset": {
+          "type": "integer",
+          "default": 2,
+          "description": "Offset for comments after a .fill, inline is 3."
```
What does "inline" mean? Why does inline not match with the default? What is this indentation relative to? The beginning, or extra tabs since the last text. Could be more clear.

porglezomp commented on src/extension.ts:
```diff
@@ -17,7 +17,7 @@ const LC2K_MODE: vscode.DocumentFilter = {
   // Use the console to output diagnostic information (console.log) and errors (console.error)
   // This line of code will only be executed once when your extension is activated
-  console.log('Congratulations, your extension "weeeee" is now active!');
+  console.log('Congratulations, your extension lc2k is now active!');
```
I feel like you don't want to leave this `console.log` in, now's a good time to delete it and the comment.

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,83 @@
+    var regex;
+    var newText;
+    var offset;
+    for (let i = 0; i < document.lineCount; i++) {
```
You need to use the line range given to you by the `range` parameter here. You can get the line number from the `Range`'s beginning position and end position.

vivianbp replied:
Should be done

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,83 @@
+        case "sw":
+        case "beq":
+          regex = /^([^\s]*)\s+(add|nor|nand|lw|sw|beq)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+(.*)?/g;
+          newText = text.replace(regex, "$1\t$2\t$3\t$4\t$5\t$6");
```
You should be careful to avoid formatting text with a trailing tab if there's no comment.

vivianbp replied:
I changed this, and I think the issue has been resolved

vivianbp replied:
I changed this, and I think the issue has been resolved

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,83 @@
+        case "lw":
+        case "sw":
+        case "beq":
+          regex = /^([^\s]*)\s+(add|nor|nand|lw|sw|beq)\s+([^\s]+)\s+([^\s]+)\s+([^\s]+)\s+(.*)?/g;
```
The final `\s+` will make this fail to match lines that have no comment in them.
Maybe try ending with:
```javascript
(?:\s+(.*))?;
```

vivianbp replied:
fixed by having whitespace be an optional capture

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,83 @@
+          break;
+
+        case "jalr":
+          regex = /^([^\s]*)\s+(jalr)\s+([^\s]+)\s+([^\s]+)\s+(.*)?/g;
```
Same issues in the other formatting cases.

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,83 @@
+    }
+
+    // last line not empty or whitespace and feature enabled //
+    const lastLine = document.lineAt(document.lineCount - 1);
```
Maybe the formatter could delete lines of whitespace that are in the middle of the document?

vivianbp replied:
It could be an option, but I think that usually if someone is putting white space lines in their document it would be temporary and for legibility?

vivianbp committed:
fixed offset settings

vivianbp committed:
Fixed optional indentation
disabled the "manual" newline at end feature
...in favor of definition in package.json
enabled range formatting (but its bad)

vivianbp committed:
Cleanup

vivianbp committed:
rebasing, ignore this

vivianbp committed:
rebasing, ignore this

porglezomp commented on src/lc2kErrorChecking:
```diff
@@ -0,0 +1,28 @@
+let diagnosticCollection: vscode.DiagnosticCollection;
```
Rename this file to have `.js`?

porglezomp commented on src/lc2kErrorChecking:
```diff
@@ -0,0 +1,28 @@
+let diagnosticCollection: vscode.DiagnosticCollection;
+
+export function activate(ctx: vscode.ExtensionContext): void {
+  ...
```
You probably shouldn't commit files which are full of placeholders 😆

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,85 @@
+    // console.log("jalr offset: " + fillOffset);
+
+    var edits = [];
+    var regex;
```
I feel like you should declare these variables inside the narrower scope.

porglezomp commented on src/lc2kFormat.ts:
```diff
@@ -1,13 +1,85 @@
+    // console.log("jalr offset: " + jalrOffset);
+    // console.log("jalr offset: " + fillOffset);
+
+    var edits = [];
```
You can use `let` instead of `var`, it has better scoping behavior. It won't make any difference in this particular case, but it's something to keep in mind.

porglezomp commented on CHANGELOG.md:
```diff
@@ -1,22 +1,41 @@
+  * null option in offset will disable it
+  * Option (default on) to put one newline at the end of the file changed to builtin 
+* Cleaned up extension.ts
+* 
```
Delete the blank line

vivianbp committed:
Almost ready to publish, worked on readme

vivianbp committed:
deleted unused code

vivianbp committed:
using let instead of var now
moved varables to inside loop

vivianbp committed:
fixed various bugs and updated changelog/readme

porglezomp commented on package.json:
```diff
@@ -55,20 +55,26 @@
           "description": "Enables this extention to format .as files."
         },
-        "lc2k.format.jalroffset": {
+        "lc2k.format.irTypeOffset": {
```
Spelled `irType` here

porglezomp commented on README.md:
```diff
@@ -34,8 +36,8 @@
     * You can disable either of these, but that's not reccomended.
-  * format.filloffset, format.jalroffset, format.noopoffset
-    * Changes the tab offset for comments for given opcodes.
+  * format.ioTypeOffset, format.jTypeOffset, format.oTypeOffset, format.fillOffset
```
Spelled `ioType` here, should be `irType`?

vivianbp committed:
Typeo and grammar
